src/main/java/com/microsoft/jenkins/kubernetes/ResourceManager.java

```diff
@@ -0,0 +1,134 @@
+            }
+            metadata = meta;
+            checkState(StringUtils.isNotBlank(getName()),
+                    Messages.KubernetesClientWrapper_noName(), getKind(), resource);
+        }
+
+        final String getNamespace() {
+            if (metadata != null) {
+                return metadata.getNamespace();
+            }
+            return Constants.DEFAULT_KUBERNETES_NAMESPACE;
+        }
+
+        final T get() {
+            return resource;
+        }
+
+        final String getName() {
+            String name = null;
+            if (metadata != null) {
+                name = metadata.getName();
+            }
+            return name;
+        }
+
+        @Deprecated
+        final String getKind() {
+            return "GetKind not support in this version";
+        }
+
+        /**
+         * Explicitly apply the configuration if a resource with the same name exists in the namespace in the cluster,
+         * or create one if not.
+         * <p>
+         * If we cannot load resource during application (possibly because the resource gets deleted after we first
+         * checked), or some one created the resource after we checked and before we created, the method fails with
+         * exception.
+         *
+         * @throws IOException if we cannot find the resource in the cluster when we apply the configuration
+         */
+        final void createOrApply() throws IOException {
+            T original = getCurrentResource();
+            T current = get();
+            T updated;
+            if (original != null) {
+                updated = applyResource(original, current);
+                if (updated == null) {
+                    throw new IOException(Messages.KubernetesClientWrapper_resourceNotFound(
+                            getKind(), CommonUtils.getResourceName(current)));
+                }
+                logApplied(updated);
+            } else {
+                updated = createResource(get());
+                logCreated(updated);
+            }
+            notifyUpdate(original, updated);
+        }
+
+        abstract T getCurrentResource();
+
+        abstract T applyResource(T original, T current);
+
+        abstract T createResource(T current);
+
+        abstract void notifyUpdate(T original, T current);
+
+        void logApplied(T res) {
+            getLogger().info(Messages.KubernetesClientWrapper_applied(res.getClass().getSimpleName(), res));
+        }
+
+        void logCreated(T res) {
+            getLogger().info(Messages.KubernetesClientWrapper_created(res.getClass().getSimpleName(), res));
+        }
+    }
+
+    public Logger getLogger() {
+        return logger;
+    }
+
+    protected void handleApiException(ApiException e) {
+        int code = e.getCode();
+        String responseBody = e.getResponseBody();
+        getLogger().error(String.format("Api call failed with code %d, detailed message: %s", code, responseBody));
```

Comment:
Move these logger messages to [Messages.properties](https://github.com/jenkinsci/kubernetes-cd-plugin/blob/dev/src/main/resources/com/microsoft/jenkins/kubernetes/Messages.properties) for localization and user friendly message maintenance

Reply:
fixed.